flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/connector/kafka/source/enumerator/KafkaSourceEnumerator.java

```diff
@@ -558,25 +533,100 @@ public PartitionOffsetsRetrieverImpl(
             }
         }
 
+        /**
+         * List offsets for the specified partitions and OffsetSpec. This operation enables to find
+         * the beginning offset, end offset as well as the offset matching a timestamp in
+         * partitions.
+         *
+         * @see KafkaAdminClient#listOffsets(Map)
+         * @param topicPartitionOffsets The mapping from partition to the OffsetSpec to look up.
+         * @return The list offsets result.
+         */
+        private Map<TopicPartition, ListOffsetsResult.ListOffsetsResultInfo> listOffsets(
+                Map<TopicPartition, OffsetSpec> topicPartitionOffsets) {
+            try {
+                return adminClient
+                        .listOffsets(topicPartitionOffsets)
+                        .all()
+                        .thenApply(
+                                result -> {
+                                    Map<TopicPartition, ListOffsetsResult.ListOffsetsResultInfo>
+                                            offsets = new HashMap<>();
+                                    result.forEach(
+                                            (tp, listOffsetsResultInfo) -> {
+                                                if (listOffsetsResultInfo != null) {
+                                                    offsets.put(tp, listOffsetsResultInfo);
+                                                }
+                                            });
+                                    return offsets;
+                                })
+                        .get();
+            } catch (InterruptedException e) {
+                throw new FlinkRuntimeException(
```

Comment:
Please also set the interrupted flag if you handle the `InterruptedException`

```java
Thread.currentThread().interrupt();
```

Reply:
Thank you for this reminder, I also find there is another `InterruptedException` ad `committedOffsets` so I revised it too.